File: Leetcode/53-Maximum-Subarray/MaximumSubarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        if (nums.size() == 0) {
            return 0;
        }
        int res = INT_MIN;
        int sum = 0;
        for (int i=0; i<nums.size(); i++) {
            sum += nums[i];
            res = max(res, sum);
            sum = max(sum, 0);
        }
        return res;
    }
};

File: Leetcode/53-Maximum-Subarray/MaximumSubarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if nums == None or len(nums) == 0 :
            return 0

        res = -sys.maxint-1
        numSum = 0
        for i in nums:
            numSum += i;
            res = max(res, numSum)
            numSum = max(numSum, 0)
        return res
        
